### PR TITLE
Bkearney/10339

### PR DESCRIPTION
**Pull request by bkearney**

### CODE REVIEW

File: lib/tasks/export_tasks.rake [hunk 1]
```diff
@@ -0,0 +1,241 @@
+#
+# export_tasks.rake is a debugging tool to extract tasks from the
+# current foreman instance.
+# 
+# Run "foreman-rake export_tasks" to export tasks whcih are not listed
+# as successful. 
+# To export all tasks "foreman-rake export_tasks tasks=all"
+
+desc 'Export Dynflow Tasks'
+  task :export_tasks => :environment do
```
Comment (iNecas) [inline lib/tasks/export_tasks.rake]: indentation -2

File: lib/tasks/export_tasks.rake [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+        puts "#{count}/#{total}"
+        count += 1
+      end
+
+      File.open(File.join(tmp_dir, "index.html"), 'w') {|file| file.write(PageHelper.pagify(PageHelper.generate_index(tasks)))}
+
+      system("tar cvzf #{export_filename} #{tmp_dir} > /dev/null")
+    end
+
+    puts "Created #{export_filename}"
```
Comment (iNecas) [inline lib/tasks/export_tasks.rake]: indentation -2

**Comment by iNecas** [comment]
The first commit should not be there

File: lib/tasks/export_tasks.rake (hunk 1 above)
Comment (iNecas) [inline lib/tasks/export_tasks.rake]: Also, it should be probably in 'foreman_tasks' namespace, as we do in https://github.com/theforeman/foreman-tasks/blob/master/lib/foreman_tasks/tasks/dynflow.rake#L1

Comment (bkearney) [inline lib/tasks/export_tasks.rake]: the indentation is at 2, whic his hte standard. yes?
Also.. on the namespace, do you want the file moved to that directory as well?

Comment (iNecas) [inline lib/tasks/export_tasks.rake]: Yes please: the directory under foreman_tasks/tasks would be better fit.
Yes: the standard is 2, but there are still too many 4-line ones, starting here https://github.com/theforeman/foreman-tasks/pull/106/files#diff-10af87cb135065f8f7ed35b14a284ccbR12

**Comment by bkearney** [comment]
@iNecas  all changes have been applied

File: lib/foreman_tasks/tasks/export_tasks.rake [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+
+      tasks.all.each do |task|
+        File.open(File.join(tmp_dir, "#{task.id}.html"), 'w') {|file| file.write(PageHelper.pagify(renderer.render_task(task)))}
+        puts "#{count}/#{total}"
+        count += 1
+      end
+
+      File.open(File.join(tmp_dir, "index.html"), 'w') {|file| file.write(PageHelper.pagify(PageHelper.generate_index(tasks)))}
+
+      system("tar cvzf #{export_filename} #{tmp_dir} > /dev/null")
```
Comment (iNecas) [inline lib/foreman_tasks/tasks/export_tasks.rake]: `s/system/sh/g` http://rake.rubyforge.org/classes/FileUtils.html#M000018 the behaviour is the same as system, but the important benefit of usnig it is, if something fails when calling the command, the user gets an error describing the problem and the rake task fails without continuing.

**Comment by iNecas** [comment]
Thanks, one last comment. Ack otherwise

**Comment by bkearney** [comment]
@iNecas in there.

**Comment by iNecas** [comment]
Thanks. Merging now

**Pull request merged**